src/extension.ts

```diff
@@ -186,12 +186,16 @@ async function activateServerForFolder(context: ExtensionContext, uri: Uri, fold
   });
 
   let serverExecutable;
+  let addInternalServerPath: string | undefined = undefined; // if we download HLS, add that bin dir to PATH
   try {
     // Try and find local installations first
     serverExecutable = findManualExecutable(logger, uri, folder) ?? findLocalServer(context, logger, uri, folder);
     if (serverExecutable === null) {
       // If not, then try to download haskell-language-server binaries if it's selected
-      serverExecutable = await downloadHaskellLanguageServer(context, logger, uri, folder);
+      await downloadGHCup(context, logger);
+      serverExecutable = await downloadHaskellLanguageServer(context, logger);
+      await getProjectGHCVersion(serverExecutable, currentWorkingDir, logger)
```

Comment:
After getting the project version, shouldn't we use it to verify that we have installed a matching HLS executable?

Reply:
Yeah

Reply:
![Screenshot_2022-02-28_12-28-19](https://user-images.githubusercontent.com/1241845/155975985-bdca303d-8323-4161-aa66-85ea547a4811.png)